omod/src/main/java/org/openmrs/module/owa/web/controller/OwaRestController.java

```diff
@@ -162,12 +164,15 @@
 				HttpSession session = request.getSession();
 				if (!url.isEmpty()) {
 					InputStream inputStream = ModuleUtil.getURLStream(downloadUrl);
-					log.warn("url pathname: " + downloadUrl.getPath());
-					String fileName = downloadUrl.getQuery().substring(downloadUrl.getQuery().lastIndexOf("=") + 1);
+					log.warn("Url pathname: " + downloadUrl.getPath());
+					String fileName = urlObject.getFileName() != null ? urlObject.getFileName() + ".zip" :
```

Comment:
As well this assumes the url will always be of this form; 
`https://bintray.com/openmrs/owa/download_file?file_path=cohortbuilder-1.0.0-beta.zip`.  Where `cohortbuilder-1.0.0-beta.zip` basically represents any module name. It however fails when the url is slightly modified to something like;
`https://bintray.com/openmrs/owa/download_file?file_path=cohortbuilder-1.0.0-beta.zip?time=today` (notice the extra url param), which is still perfectly ok.